### PR TITLE
Add fingerprint protection module

### PR DESCRIPTION
**Pull request by joshliebe**
Task/Issue URL: https://app.asana.com/0/488551667048375/1203179499015587/f

### Description
This PR adds:
- Fingerprint protection module
- Feature plugins and content scope plugins for each feature (Battery, canvas, hardware, screen size and temporary storage)

### Steps to test this PR
- Freshly install the app from this branch
- In `fingerprint_protection.db`, verify that the `fingerprinting_battery`, `fingerprinting_canvas`, `fingerprinting_hardware`, `fingerprinting_screen_size` and `fingerprinting_temporary_storage` tables are populated with JSON.

### CODE REVIEW

**Comment by joshliebe** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2523** <a href="https://app.graphite.dev/github/pr/duckduckgo/android/2523" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/android/2523?utm_source=stack-comment).

File: fingerprint-protection/fingerprint-protection-impl/src/main/java/com/duckduckgo/fingerprintprotection/impl/di/FingerprintProtectionModule.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+@Module
+@ContributesTo(AppScope::class)
+object FingerprintProtectionModule {
+
+    @SingleInstanceIn(AppScope::class)
+    @Provides
+    fun provideFingerprintProtectionDatabase(context: Context): FingerprintProtectionDatabase {
+        return Room.databaseBuilder(context, FingerprintProtectionDatabase::class.java, "fingerprint_protection.db")
+            .enableMultiInstanceInvalidation()
```
Comment (anikiki) [inline fingerprint-protection/fingerprint-protection-impl/src/main/java/com/duckduckgo/fingerprintprotection/impl/di/FingerprintProtectionModule.kt:50]: Do we need to enable multi instance invalidation? See  https://app.asana.com/0/488551667048375/1203276170799080/f

**Review by anikiki — APPROVED** [review]
Added a question about the use of `enableMultiInstanceInvalidation()`.
Looks good and works as expected! :tada:

File: fingerprint-protection/fingerprint-protection-impl/src/main/java/com/duckduckgo/fingerprintprotection/impl/di/FingerprintProtectionModule.kt (hunk 1 above)
Comment (joshliebe) [reply fingerprint-protection/fingerprint-protection-impl/src/main/java/com/duckduckgo/fingerprintprotection/impl/di/FingerprintProtectionModule.kt:50]: Good spot @anikiki! I'll remove

**Pull request merged**